web/web.go

```diff
@@ -82,6 +82,25 @@ func secureHeadersMiddleware(h http.Handler) http.Handler {
 	})
 }
 
+// withStackTrace logs the stack trace in case the request panics. The function
+// will re-raise the error which will then be handled by the net/http package.
+// It is needed because the go-kit log package doesn't manage properly the
+// panics from net/http (see https://github.com/go-kit/kit/issues/233).
+func withStackTracer(h http.Handler, l log.Logger) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				const size = 64 << 10
```

Comment:
do you think this might be causing the `failed to restore the stack` ?

Reply:
I'm not sure, this code is taken straight from the [net/http](https://github.com/golang/go/blob/4059afb41925a2ac42a3c840b9d88ac08dd01c99/src/net/http/server.go#L1723-L1728) package. Will try to reproduce locally...

Reply:
I couldn't determine if this is causing the missing stack...